cmd/syncthing/gui.go

```diff
@@ -306,16 +290,15 @@ func (s *apiService) Serve() {
 	// The main routing handler
 	mux := http.NewServeMux()
 	mux.Handle("/rest/", restMux)
+
 	mux.HandleFunc("/qr/", s.getQR)
 
 	// Serve compiled in assets unless an asset directory was set (for development)
-	mux.Handle("/", s.statics)
+	mux.Handle("/", newStaticsServer(guiCfg.Theme, s.assetDir))
```

Comment:
You could have map[config.GUIConfiguration]staticServer (or some other better key for the map) map on the API service which calls setTheme on the right one as it detects the change. Or to be honest it could just call it on every one every config save, as all that changes is just a string.

Reply:
Here again, maybe we can use the `Address` as key for the `staticServer` map.

Reply:
Sure